client/meter/blockchain.js

```diff
@@ -193,6 +227,35 @@ async function autoBuy(amount) {
   return Promise.all(promises)
 }
 
+async function updateBlock(blockNumber) {
+  if (blockNumber < lastCount) return;
+  const block = await bonds.findBlock(blockNumber);
+  const txs = block.transactions;
+  for (const tx of txs) {
+    const txDetail = await bonds.transaction(tx);
+    const to = txDetail.to;
+    if (to === ENERGY_MASTER_ADDRESS) {
+      console.log("Updating master");
+      await updateMaster();
+    } else if (contracts[to] !== undefined) {
+      console.log(`Updating contract ${to}`);
+      await getContractInfoByAddress(to);
+    }
+  }
+  console.log(`Synced block ${blockNumber}`);
+  lastBlock = blockNumber;
+}
+
+async function updateBlockchain() {
+  if (!inited) await init();
+  const blockNumber = await bonds.blockNumber;
+  for (let i = lastBlock + 1; i <= blockNumber; i++) {
```

Comment:
We can update all at the same time, there should be no dependencies, as we update everything to latest instead of salvage details of tx and use that to update.